src/averell/readers/disco3.py

```diff
@@ -1,9 +1,135 @@
+import re
+import xml.etree.ElementTree as ETree
 from pathlib import Path
 
 XML_PATH = Path("tei") / "all-periods-per-author"
 
+def parse_xml(xml_file):
+    """
+    Disco-3 corpus parser
+    :param xml_file: Path of the corpus file
+    :return:
+    """
+    tree = ETree.parse(xml_file)
+    root = tree.getroot()
+    line_group_list = root.findall(
+        ".//*{http://www.tei-c.org/ns/1.0}head/..[@type]/..{http://www.tei-c.org/ns/1.0}lg")
+
+    author = root.find(".//{http://www.tei-c.org/ns/1.0}author")
+    author = re.sub(r"[\n ]+", " ", author.text)
+
+    language = root.find(".//{http://www.tei-c.org/ns/1.0}language")
+    language = re.sub(r"[\n ]+", " ", language.text)
+
+    sonnet_list = []
+    id_list = []
+    is_ensemble = False
+    ensemble_dict = {}
+
+    for line_group in line_group_list:
+        if line_group.get("type") == "sonnet-sequence":
+            is_ensemble = True
+            ensemble_title = line_group.find(
+                '{http://www.tei-c.org/ns/1.0}head')
+            ensemble_title = "".join(ensemble_title.itertext())
+            ensemble_id = line_group.get(
+                "{http://www.w3.org/XML/1998/namespace}id")
+            ensemble_dict.update({
+                'ensemble_id': ensemble_id,
+                'ensemble_title': ensemble_title,
+            })
+            ensemble_alt_title = root.find(
+                f".//*[@resource='disco:{ensemble_id}']"
+                + "/{http://www.tei-c.org/ns/1.0}title[@property='dc:alternative']")
+            if ensemble_alt_title is not None:
+                ensemble_alt_title = re.sub(r"[\n ]+", " ", "".join(
+                    ensemble_alt_title.itertext()))
+                ensemble_dict.update({
+                    'ensemble_alt_title': ensemble_alt_title
+                })
+            line_group.clear()
+        else:
+            work_dict = {}
+            if line_group.get("type"):
+                sonnet_id = line_group.get(
+                    "{http://www.w3.org/XML/1998/namespace}id")
+                work_dict.update({
+                    'poem_id': sonnet_id,
+                    'poem_type': line_group.get("type"),
+                    'author': author,
+                    'language': language,
+                })
+                id_list.append(work_dict["poem_id"])
+                alt_title = root.find(
+                    f".//*[@resource='disco:{sonnet_id}']"
+                    + "/{http://www.tei-c.org/ns/1.0}title[@property='dc:alternative']")
+                if alt_title is not None:
+                    alt_title = re.sub(r"[\n ]+", " ",
```

Comment:
Better if you do 
`alt_title = re.sub(
                        r"[\n ]+", " ",`

Reply:
Done